pkg/cloudprovider/types.go

```diff
@@ -257,6 +257,31 @@ func (ofs Offerings) Cheapest() Offering {
 	})
 }
 
+// MostExpensive returns the most expensive offering from the return offerings
+func (ofs Offerings) MostExpensive() Offering {
+	return lo.MaxBy(ofs, func(a, b Offering) bool {
+		return a.Price > b.Price
+	})
+}
+
+// CapacityType returns the capacity type requirement for an offering, and panics if a capacity type is not set
+func (of Offering) CapacityType() string {
+	req, ok := of.Requirements[v1beta1.CapacityTypeLabelKey]
+	if !ok {
+		panic("failed to get capacity type for offering, capacity type must be definied on all offerings")
+	}
+	return req.Values()[0]
+}
+
+// Zone returns the zonal requirement for an offering, and panics if a zone is not set
+func (of Offering) Zone() string {
+	req, ok := of.Requirements[v1.LabelTopologyZone]
+	if !ok {
+		panic("failed to get zone for offering, zone must be definied on all offerings")
```

Comment:
same here on panic and nit
```suggestion
		panic("failed to get zone for offering, zone must be defined on all offerings")
```